mlir/lib/Conversion/ArithToEmitC/ArithToEmitC.cpp

```diff
@@ -288,6 +288,29 @@ class CmpIOpConversion : public OpConversionPattern<arith::CmpIOp> {
   }
 };
 
+class NegFOpConversion : public OpConversionPattern<arith::NegFOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(arith::NegFOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    auto adaptedOp = adaptor.getOperand();
+    auto adaptedOpType = adaptedOp.getType();
+
+    if (!isa<FloatType>(adaptedOpType)) {
+      return rewriter.notifyMatchFailure(op.getLoc(),
+                                         "negf currently only supported on "
+                                         "floats, not tensors/vectors thereof");
+    }
+
```

Comment:
nit: Check with `isSupportedFloatType` and fail if the bitwidth is not supported.

Reply:
Very good point, I adapted the check.

Reply:
Can you update the error message as well?

Reply:
What should it mention? Should it explicitly list the supported float types? Or just mention that the reason could be an unsupported float type?

Reply:
Yes, the error is confusing when the type is a scalar f80 for example.

Reply:
I split it into two checks.